cmd/collector/app/server/grpc.go

```diff
@@ -42,31 +40,19 @@ type GRPCServerParams struct {
 // StartGRPCServer based on the given parameters
 func StartGRPCServer(params *GRPCServerParams) (*grpc.Server, error) {
 	var server *grpc.Server
-	var grpcOpts []grpc.ServerOption
-
-	if params.MaxReceiveMessageLength > 0 {
-		grpcOpts = append(grpcOpts, grpc.MaxRecvMsgSize(params.MaxReceiveMessageLength))
-	}
-	grpcOpts = append(grpcOpts, grpc.KeepaliveParams(keepalive.ServerParameters{
-		MaxConnectionAge:      params.MaxConnectionAge,
-		MaxConnectionAgeGrace: params.MaxConnectionAgeGrace,
-	}))
-
-	if params.TLSConfig.Enabled {
-		// user requested a server with TLS, setup creds
-		tlsCfg, err := params.TLSConfig.Config(params.Logger)
-		if err != nil {
-			return nil, err
-		}
-
-		creds := credentials.NewTLS(tlsCfg)
-		grpcOpts = append(grpcOpts, grpc.Creds(creds))
+	var grpcOpts []configgrpc.ToServerOption
+	server, err := params.ToServer(context.Background(), nil, component.TelemetrySettings{
+		Logger: params.Logger,
+		LeveledMeterProvider: func(_ configtelemetry.Level) metric.MeterProvider {
+			return noop.NewMeterProvider()
+		},
+	}, grpcOpts...)
+	if err != nil {
+		return nil, err
 	}
-
-	server = grpc.NewServer(grpcOpts...)
 	reflection.Register(server)
 
-	listener, err := net.Listen("tcp", params.HostPort)
```

Comment:
I think your tests are failing because `tcp` isn't being explicitly set anymore. We'll want to explictly set the `Trasnport` to `TransportTypeTCP` in the grpc server configuration